tests/drop_packets/drop_packets.py

```diff
@@ -258,6 +261,107 @@ def ports_info(ptfadapter, duthosts, rand_one_dut_hostname, setup, tx_dut_ports)
     return data
 
 
+def acl_setup(duthosts, loganalyzer, template_dir, acl_rules_template, del_acl_rules_template, dut_tmp_dir,
+              dut_clear_conf_file_path):
+    for duthost in duthosts:
+        acl_facts = duthost.acl_facts()["ansible_facts"]["ansible_acl_facts"]
+        if 'DATAACL' not in acl_facts.keys():
+            pytest.skip("Skipping test since DATAACL table is not supported on this platform")
+
+        duthost.command("mkdir -p {}".format(dut_tmp_dir))
+        dut_conf_file_path = os.path.join(dut_tmp_dir, acl_rules_template)
+
+        logger.info("Generating config for ACL rule, ACL table - DATAACL")
+        duthost.template(src=os.path.join(template_dir, acl_rules_template), dest=dut_conf_file_path)
+        logger.info("Generating clear config for ACL rule, ACL table - DATAACL")
+        duthost.template(src=os.path.join(template_dir, del_acl_rules_template), dest=dut_clear_conf_file_path)
+
+        logger.info("Applying {}".format(dut_conf_file_path))
+
+        loganalyzer[duthost.hostname].expect_regex = [LOG_EXPECT_ACL_RULE_CREATE_RE]
+        with loganalyzer[duthost.hostname]:
+            duthost.command("config acl update full {}".format(dut_conf_file_path))
+
+
+def acl_teardown(duthosts, loganalyzer, dut_tmp_dir, dut_clear_conf_file_path):
+    for duthost in duthosts:
+        loganalyzer[duthost.hostname].expect_regex = [LOG_EXPECT_ACL_RULE_REMOVE_RE]
+        with loganalyzer[duthost.hostname]:
+            logger.info("Applying {}".format(dut_clear_conf_file_path))
+            duthost.command("config acl update full {}".format(dut_clear_conf_file_path))
+            logger.info("Removing {}".format(dut_tmp_dir))
+            duthost.command("rm -rf {}".format(dut_tmp_dir))
+            time.sleep(ACL_COUNTERS_UPDATE_INTERVAL)
+
+
+@pytest.fixture
+def acl_ingress(duthosts, loganalyzer):
+    """ Create acl rule defined in config file. Delete rule after test case finished """
+    base_dir = os.path.dirname(os.path.realpath(__file__))
+    template_dir = os.path.join(base_dir, 'acl_templates')
+    acl_rules_template = "acltb_test_rule.json"
+    del_acl_rules_template = "acl_rule_del.json"
+    dut_tmp_dir = os.path.join("tmp", os.path.basename(base_dir))
+    dut_clear_conf_file_path = os.path.join(dut_tmp_dir, del_acl_rules_template)
+
+    acl_setup(duthosts, loganalyzer, template_dir, acl_rules_template, del_acl_rules_template, dut_tmp_dir,
+              dut_clear_conf_file_path)
+    yield
+    acl_teardown(duthosts, loganalyzer, dut_tmp_dir, dut_clear_conf_file_path)
+
+
+def create_or_remove_acl_egress_table(duthost, setup, op):
+    acl_table_config = {
+        "table_name": "OUTDATAACL",
+        "table_ports": ",".join(duthost.acl_facts()["ansible_facts"]["ansible_acl_facts"]["DATAACL"]["ports"]),
+        "table_stage": "egress",
+        "table_type": "L3"
+    }
+
+    for sonic_host_or_asic_inst in duthost.get_sonic_host_and_frontend_asic_instance():
+        if op == "add":
```

Comment:
Please consider adding a verification to check if ACL table was created successfully. Two reasons:
1. EGRESS ACL is not supported on all platforms
2. We have limited ACL resources, which may result in creaing ACL table failed.
Thanks

Reply:
Thank you,
I added a loganalyzer check that will check that it successfully added the acl table, could you look at it?

I took this approach from the test_acl.py file, so I hope this check I added now is sufficient.